File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;
int a,b; //가로 세로
int solution(vector<vector<int>> sizes) {
    int answer = 0;
    
    for(auto s : sizes){
        if(s[0] < s[1]){
            a = max(a,s[1]);
            b = max(b,s[0]);
        }
        else{    
            a = max(a,s[0]);
            b = max(b,s[1]);
        }
    }
    
    answer = a * b;
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    int a,b;
    public int solution(int[][] sizes) {
        for(int[] size : sizes){
            a = Math.max(a, Math.max(size[0],size[1]));
            b = Math.max(b, Math.min(size[0],size[1]));
        }
        return a * b;
    }
}